rust/bin-shared/src/tun_device_manager/windows.rs

```diff
@@ -316,7 +338,7 @@ impl tun::Tun for Tun {
 // Moves packets from Internet towards the user
 fn start_send_thread(
     mut packet_rx: mpsc::Receiver<IpPacket>,
-    session: Arc<wintun::Session>,
+    session: Weak<wintun::Session>,
```

Comment:
Oh, so is this like using a `{ [weak self] in }` in Swift?

Reply:
Yeah it is a weak-pointer to an `Arc`, `Arc` being the strong-pointer. Rust enforces this in the type-system.